### PR TITLE
dd-agent: remove

### PR DESCRIPTION
**Pull request by armeenm**
###### Description of changes
This removes `dd-agent` (v5) and its corresponding NixOS module. `dd-agent` has been superseded by `datadog-agent` as briefly explained [here](https://github.com/DataDog/datadog-agent#datadog-agent). This is being done as `dd-agent` [depends on python2](https://github.com/DataDog/dd-agent#setup-your-environment).
This is my first NixOS module contribution, so please don't hesitate to let me know if I should be doing something differently here.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by delroth** [comment]
This should probably be accompanied by a release notes change. It might be worth providing a simple example of config migration from dd-agent to datadog-agent if that's something that can easily be done.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -283,6 +283,7 @@ mapAliases ({
   dbus_libs = throw "'dbus_libs' has been renamed to/replaced by 'dbus'"; # Converted to throw 2022-02-22
   dbus_tools = throw "'dbus_tools' has been renamed to/replaced by 'dbus.out'"; # Converted to throw 2022-02-22
   dbvisualizer = throw "dbvisualizer has been removed from nixpkgs, as it's unmaintained"; # Added 2020-09-20
+  dd-agent = throw "dd-agent has been removed in favor of the newer datadog-agent"; # Added 2022-04-26
```
Comment (delroth) [inline pkgs/top-level/aliases.nix:286]: Similar to this, you'll want something in nixos/modules/... using mkRemovedOptionModule to give a better error message when configs are using the old services.dd-agent options. Either put it in the datadog-agent module .nix file, or in rename.nix with other deprecated modules (I don't know which of the two is more idiomatic, I suspect either is fine).

**Comment by Sohalt** [comment]
@armeenm will you find the time to implement the `mkRemovedOptionModule`?

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (Sohalt) [reply pkgs/top-level/aliases.nix:286]: I've added the `mkRemovedOptionModule` and rebased in https://github.com/NixOS/nixpkgs/pull/187857.

**Comment by domenkozar** [comment]
Would be great to get this done:)

**Comment by domenkozar** [comment]
Fixed in master

**Pull request closed without merging**